File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.py
def solution(n):
    answer: int = 1
    
    sum_num: int = 0
    for i in range(1, n-1):
        sum_num += i
        for j in range(i + 1, n):
            sum_num += j
            if sum_num == n:
                answer += 1
                sum_num = 0
                break
            elif sum_num > n:
                sum_num = 0
                break
    
    return answer

File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.java
class Solution {
    public int solution(int n) {
        int answer = 0;
        
        for (int i = 1; i <= n; i++) {
            int number = 0;
            for (int j = i; j <= n; j++) {
                number += j;
                if (number == n) {
                    answer ++;
                    break;
                }
                else if (number > n)
                    break;
            }
        }
        
        return answer;
    }
}